### PR TITLE
Docsify Based Markdown Documentation

### PR DESCRIPTION
**Pull request by Shubhaankar-Sharma**
### Preview :- [Docs](https://shubhaankar-sharma.github.io/chi)
Let me know what to change...
This is supposed to be served with github pages, where /docs is the directory where index.html is placed

### CODE REVIEW

**Comment by VojtechVitek** [comment]
Lovely!

@pkieltyka Do we want to create a separate repo for the docs? Or perhaps we could use a dedicated branch for the docs?

**Comment by Shubhaankar-Sharma** [comment]
Yea I think we can do a orphan branch for docs as well...

**Comment by pkieltyka** [comment]
@VojtechVitek @Shubhaankar-Sharma what if we merged to master under “docs/“ folder? btw I also registered go-chi.io for us to put the docs site on :)

**Comment by Shubhaankar-Sharma** [comment]
@pkieltyka yes thats what i originally planned... so now the import path will also be go-chi.io/chi/v5 ? awesome

**Comment by pkieltyka** [comment]
I’m going to keep the import path the same for now, but in the future will consider a vanity import path. For now go-chi.io will just be for the docs site

**Comment by Shubhaankar-Sharma** [comment]
ok cool 👍 then we should just do in /docs in my opinion...

File: docs/user_guide/testing.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+	rr := httptest.NewRecorder()
+	s.Router.ServeHTTP(rr, req)
+
+	return rr
+}
+
+// checkResponseCode is a simple utility to check the response code
+// of the response
+func checkResponseCode(t *testing.T, expected, actual int) {
+	if expected != actual {
```
Comment (robsonpeixoto) [inline docs/user_guide/testing.md:89]: Please, add `t.Helper()`

Comment (Shubhaankar-Sharma) [reply docs/user_guide/testing.md:89]: Will do 👍

**Pull request merged**